internal/page/page.go

```diff
@@ -74,3 +79,33 @@ func (o limitOffset) Int64() int64 {
 func (o limitOffset) String() string {
 	return strconv.Itoa(int(o))
 }
+
+type Metadata struct {
+	Previous string `json:"previous,omitempty"`
+	Next     string `json:"next,omitempty"`
+}
+
+func MetadataResponse(ctx context.Context, r *http.Request, total int) *Metadata {
```

Comment:
Am I missing where we are ommiting the next link when it's the last page?
Or how do we approach the last and first page? I thought we would have the next and prev missing, right?

Reply:
So I thought to show the next as the minimux between (limit +offset) and the total records. 
but I might be wrong here. 
btw, sources are not showing an empty string when there is no next page.

Reply:
I'm not sure we want to strictly follow exactly what sources are doing :innocent:
I'm not sure what would be best approach, but empty link sounds as a good approach to me, it's easy to recognize for client that there are no more results IMO :shrug:
As we do not have total nor page count, it is impossible to recognize for client, they'd need to do the last request for empty page, which sounds like a waste of an API call

Reply:
I wanted to follow common practice. 
IB and sources do not show an empty link, for example: 
```
  "links": {
    "first": "/api/image-builder/v1.0/composes?offset=0&limit=100",
    "last": "/api/image-builder/v1.0/composes?offset=4&limit=100"
  },
  "meta": {
    "count": 5
  }

```
And If I am talking about common practices I would change it to first and last like both of them expose.

Reply:
Seeing something in a different project does not make anything a common practice :-D

I feel like returning empty links makes life easier for the HTTP client (our customer), so little bit of extra effort might be worthwhile.

Reply:
ok I changed it to empty links if we are listing the first page and the last one.